client/src/reducers/controls.js

```diff
@@ -122,6 +121,46 @@ function createGenesDimMap(userDefinedGenes, diffexpGenes, world, crossfilter) {
   };
 }
 
+function pruneVarDataCache(varData, needed) {
+  /*
+  Remove any unneeded columns from the varData dataframe.  Will only
+  prune / remove if the total column count exceeds VarDataCacheLowWatermark
+
+  Note: this code leverages the fact that dataframe offsets indicate
+  the order in which the columns were added.  This crudely provides
+  LRU semantics, so we can delete "older" columns first.
+  */
+
+  /*
+  VarDataCacheLowWatermark - this cofig value sets the minimum cache size,
+  in columns, below which we don't throw away data.
+
+  The value should be high enough so we are caching the maximum which will
+  "typically" be used in the UI (currently: 10 for diffexp, and N for user-
+  specified genes), and low enough to account for memory use (any single
+  column size is 4 bytes * numObs, so a column can be multi-megabyte in common
+  use cases).
+  */
+  const VarDataCacheLowWatermark = 32;
```

Comment:
What do you think about putting this value in globals (or maybe a constants file?) for ease of discovery and tweaking.

Reply:
When a constant/config is used only in one place, I prefer them to be with the code they parameterize.    For me, moving them outside of the context in which they are used makes it harder to understand.    The counter example is a widely used constant/config - those should be centralized.

I realized this is an opinion.  If we establish a standard, I'm happy to follow it.